trio/_subprocess.py

```diff
@@ -568,12 +626,23 @@ async def my_deliver_cancel(process):
 
     if isinstance(stdin, str):
         raise UnicodeError("process stdin must be bytes, not str")
-    if stdin == subprocess.PIPE:
-        raise ValueError(
-            "stdin=subprocess.PIPE doesn't make sense since the pipe "
-            "is internal to run_process(); pass the actual data you "
-            "want to send over that pipe instead"
-        )
+    if task_status is trio.TASK_STATUS_IGNORED:
+        if stdin is subprocess.PIPE:
+            raise ValueError(
+                "stdin=subprocess.PIPE doesn't make sense without "
+                "nursery.start, since there's no way to access the "
+                "pipe; pass the data you want to send or use nursery.start"
+            )
+        if options.get("stdout") is subprocess.PIPE:
+            raise ValueError(
+                "stdout=subprocess.PIPE doesn't make sense without "
+                "nursery.start, since there's no way to access the pipe"
+            )
+        if options.get("stderr") is subprocess.PIPE:
+            raise ValueError(
+                "stderr=subprocess.PIPE doesn't make sense without "
+                "nursery.start, since there's no way to access the pipe"
+            )
```

Comment:
I would say "is only valid with nursery.start, which provides access the pipe".  Fewer negations is more understandable!